### ISSUE TITLE
MuonGun produces inconsistent weights between different versions (Trac #2179)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2171]
The MuonGun weighting module produces wrong weights for events above 10^6^ GeV. This behaviour starts at rev 159352 and is probably at least related to tickets #2131 and #2149.
This can be tested with running the script

```text
/home/mmeier/test_muongun_weighting.py --out_file=/path/to/file.hd5
```
once with a combo metaproject with MuonGun rev 159351 and with rev 159352 or later.

Migrated from https://code.icecube.wisc.edu/ticket/2179
```json
{
    "status": "closed", 
    "changetime": "2018-08-03T13:43:03", 
    "description": "The MuonGun weighting module produces wrong weights for events above 10^6^ GeV. This behaviour starts at rev 159352 and is probably at least related to tickets #2139 and #2157.\nThis can be tested with running the script\n{{{\n/home/mmeier/test_muongun_weighting.py --out_file=/path/to/file.hd5\n}}}\nonce with a combo metaproject with MuonGun rev 159351 and with rev 159352 or later.", 
    "reporter": "mmeier", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1533303783524855", 
    "component": "analysis", 
    "summary": "MuonGun produces inconsistent weights between different versions", 
    "priority": "normal", 
    "keywords": "MuonGun", 
    "time": "2018-08-02T15:15:04", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180802T15:24:12: olivas commented:

> Seems there's a lot going on at r159352.
> http://code.icecube.wisc.edu/projects/icecube/changeset/159352/IceCube/projects
>

**Comment by icecube-trac** [comment]
Trac update at 20180803T13:42:02: 
* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"
* jvansanten changed owner from "" to "jvansanten"
* jvansanten commented:

> In changeset 164578:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="164578"
> Set the energy bounds of SplineEnergyDistribution to the support of the backing spline in ctor, rather than default 1 GeV to 1 PeV. Fixes #2171.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20180803T13:43:03: mmeier commented:

> http://code.icecube.wisc.edu/projects/icecube/changeset/159352/IceCube/projects/MuonGun/trunk/private/MuonGun/EnergyDistribution.cxx
> 
> The additional check added in lines 98 - 101 returns -inf for all energies outside E_min_ and E_max_. This leads to ignoring all muons outside those borders in `WeightCalculator.cxx` in line 46. 10^6^ GeV is the default E_max_ for all models so im not sure if the new behavior is intended or not. At least if it's intended for all those flux models implemented in MuonGun to not produce trustworthy weights outside of those default energy borders the weight should be either 0 or nan.
